Add passwordless guide (#68)
FusionAuth/fusionauth-site: site/docs/v1/tech/guides/passwordless.adoc

When customizing, you can use any Apache FreeMarker built-ins within the template. Make sure you modify both the HTML and text templates. Here are the default templates:

.The default passwordless HTML template
[source,text]
----
include::docs/src/email/passwordless.html[] 
----

.The default passwordless text template
[source,text]
----
include::docs/src/email/passwordless.txt[]
----

You can localize the template as well:

image::guides/passwordless/localization-screen-for-email-template.png[The localization screen for your email templates.,width=700,role=box-shadow margin-right]

Here is more information about link:/docs/v1/tech/email-templates[email templates]. 

=== One time code customization

You can modify the lifetime of the code delivered to users. By default it is 180 seconds; change this in the tenant settings:

image::guides/passwordless/tenant-settings.png[The tenant settings to customize code lifetime and generation strategy.,width=700,role=box-shadow margin-right]

You can also change the type of the code generated. For example, you may want your code to be only alphanumeric characters.

> You may change your code length or generation strategy for security or user experience reasons. You may have requirements that specify a certain code length. If you deliver the code by text message, having a user enter a six digit alphanumeric code sent to them by SMS is a lot easier than a 64 byte string. 

Reviewer: ```suggestion
You may change your code length or generation strategy for security or user experience reasons. You may have requirements that specify a certain code length. If you deliver the code by text message, having a user enter a six-digit alphanumeric code sent to them by SMS is a lot easier than a 64-byte string. 
```

You have the following options for the code generation strategy:

* alphabetic characters
* alphanumeric characters
* bytes
* digits

Consult the link:/docs/v1/tech/apis/tenants[tenant API documentation] for the length limits, which vary based on the strategy.

== Security

> With passwordless authentication, if the user's email account is hijacked, their account on your system is compromised. However, many organizations have security policies and protections around email accounts. It is often easier to protect and regularly change one email account password than to change all of a user's passwords. Email accounts are also more likely to have two factor authentication enabled.

Reviewer: ```suggestion
With passwordless authentication, if the user's email account is hijacked, their account on your system is compromised. However, many organizations have security policies and protections around email accounts. It is often easier to protect and regularly change one email account password than to change all of a user's passwords. Email accounts are also more likely to have two-factor authentication enabled.
```

One way to increase the security of your passwordless authentications is to decrease the lifetime of the code. This will help if the email is compromised or accidentally forwarded. 

There are no limits on how many passwordless requests can be made for a user, but only the most recent code is valid. Using any of the others, even if they have not yet expired, will display an `Invalid login credentials` message to the user.

If someone tries to log in with an email that is not present in the FusionAuth user database, they'll see the same notification as they would if the email existed. No email will be sent.

> If you use the passwordless API, follow the principle of least privilege and limit the calls to which the API key has access. If you are using the API key only for passwordless login, then don't give this key any other permissions.

Reviewer: ```suggestion
If you use the passwordless API, follow the principle of least privilege, and limit the calls to which the API key has access. If you are using the API key only for passwordless login, then don't give this key any other permissions.
```

=== What about user's passwords

> When FusionAuth is your user datastore, adding a user means you must either provide a password or send them a link to setup their initial password.

Reviewer: ```suggestion
When FusionAuth is your user datastore, adding a user means you must either provide a password or send them a link to set up their initial password.
```

> If you are only allowing passwordless authentication for your application, don't allow the user to specify a password and instead generate a random series of characters for the password. We recommend generating at least 32 characters in the ascii character set that are completely random in order to ensure the user's account is secure.

Reviewer: ```suggestion
If you are only allowing passwordless authentication for your application, don't allow the user to specify a password and instead generate a random series of characters for the password. We recommend generating at least 32 characters in the ascii character set that are completely random to ensure the user's account is secure.
```
